### ISSUE TITLE
Stripe payment form requires two clicks to submit form

### ISSUE THREAD

**Issue by croxton** [solspace/craft3-freeform#239]
**Describe the bug or issue you're experiencing**
The submit button must be clicked twice to submit a payment form and complete the payment. 

The first click triggers a number of `xhr` requests to Stripe: 
- 3 requests to `https://r.stripe.com/0` 
- 1 request to `https://api.stripe.com/v1/payment_methods`.

The payment_methods requests returns json like this using a test card `4242424242424242`:

```js
{
  "id": "12345678abc",
  "object": "payment_method",
  "billing_details": {
    "address": {
      "city": null,
      "country": null,
      "line1": null,
      "line2": null,
      "postal_code": null,
      "state": null
    },
    "email": null,
    "name": null,
    "phone": null
  },
  "card": {
    "brand": "visa",
    "checks": {
      "address_line1_check": null,
      "address_postal_code_check": null,
      "cvc_check": null
    },
    "country": "US",
    "exp_month": 2,
    "exp_year": 2022,
    "funding": "credit",
    "generated_from": null,
    "last4": "4242",
    "networks": {
      "available": [
        "visa"
      ],
      "preferred": null
    },
    "three_d_secure_usage": {
      "supported": true
    },
    "wallet": null
  },
  "created": 1637227491,
  "customer": null,
  "livemode": false,
  "type": "card"
}
```

But nothing else happens. I presume on first click, Stripe validates the payment card, but the form does not continue with the Freeform submission process.

The second click will complete the form submission. I am listening to the `freeform-ajax-before-submit` event to demonstrate that the freeform submission process only gets triggered on the second click.  

**Steps to reproduce**

1. Create a Stripe payment form following the instructions here:
https://docs.solspace.com/craft/freeform/v3/integrations/payments/

2. Relevant form behaviour settings: 
   • Disable Submit Button on Form Submit? `off`
   • Freeform Script Insertion Location `None` (added manually)
   • Freeform Session Context: `PHP Sessions` or `Encrypted Payload` (makes no difference)

3. Template code:

```twig
{% set form = craft.freeform.form('eventBookingForm') %}
{{ form.render( {
    id: 'event-booking-form',
    overrideValues: {
        cost: entry.cost,
        event: entry.title ~ ' - ' ~ entry.eventDate|date('long')
    }, 
    returnUrl: "{{ siteUrl }}training/reservation/"~entry.slug~"/{{ submission.id }}/{{ submission.token }}"
}) }}
{{ craft.freeform.loadFreeformPlugin() }}
```             

4. Example js:
```js
const form = document.getElementById(formId);
 
form.addEventListener("freeform-ajax-before-submit", (event) => {
   // will only be triggered on second attempt to submit form
    console.log("freeform-ajax-before-submit");     
    event.preventDefault();
});    
```    

5. Complete all form fields correctly and click the submit button. Note that in my form template, I am using a `<button type="submit">` not an `<input type="submit">`

**Expected behavior**
The form should not need to be submitted twice.

**Craft & Plugin Info**

- Craft Version: 3.7.20
- Freeform Version: [dev-master](https://github.com/solspace/craft3-freeform/commit/1078456d32739f89e7f44377494bf9a2e4ff4ccf)
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by croxton** [comment]
Looks like submit buttons might need a `data-freeform-action=""` attribute, since 8b73811c502e4f4c07f0ee1a0c04893adb6ec516 ?

https://github.com/solspace/craft3-freeform/blob/ef26b0fce485533b2ef689a95bbad8f76378c5ec/packages/scripts/src/components/front-end/plugin/freeform.js#L738-L740

**Comment by croxton** [comment]
Yes that was it. I added `data-freeform-action="submit"` to my form submit button and now the form submission completes on the first click.

Maybe best to either document that requirement or change the line back to pick up all submit buttons, since I imagine you might be breaking a lot of existing forms that use custom form templates?

```js
_getSubmitButtons = () => { 
   return this.form.querySelectorAll(`*[type=submit]`); 
 };
```

**Comment by gustavs-gutmanis** [comment]
A very good catch @croxton, thanks!
I'll make it backwards compatible.

**Comment by kjmartens** [comment]
This is now fixed in Freeform 3.12.5. 🙂
